app.js

```diff
@@ -220,6 +246,27 @@ app.get('/api/secure/user', function(req, res) {
     });
 });
 
+//returns the location assosciated with the given user id
+app.get('/api/:userId/location', (req,res) => {
+  const userId = req.params.userId;
+  log.info(userId);
+  db.getLocationById(userId)
+    .then(location => res.send(location).end());
```

Comment:
We are missing a `catch` here.  Without it, any errors are lost and the user never gets back a response.